dll/win32/shell32/shlexec.cpp

```diff
@@ -2383,14 +2421,63 @@ ShellExecuteExA(LPSHELLEXECUTEINFOA sei)
 
 /*************************************************************************
  * ShellExecuteExW                [SHELL32.293]
- *
  */
 BOOL
 WINAPI
 DECLSPEC_HOTPATCH
 ShellExecuteExW(LPSHELLEXECUTEINFOW sei)
 {
-    return SHELL_execute(sei, SHELL_ExecuteW);
+    HRESULT hrCoInit;
+    DWORD dwError;
+    ULONG fOldMask;
+
+    hrCoInit = SHCoInitializeAnyApartment();
```

Comment:
MSDN says "COM should be initialized before ShellExecuteEx is called"?

Reply:
A rude application might run without COM initialization.

Reply:
But then since such app doesn't respect the documentation, it can break.
(I'm curious to know whether there exists such "rude" apps out there that don't call the COM init before ShellExecuteEx)

Reply:
> I'm curious to know whether there exists such "rude" apps out there that don't call the COM init before ShellExecuteEx

I'm sure there are many but only things such as SEE_..IDLIST or [IShellExecuteHook](https://devblogs.microsoft.com/oldnewthing/20080910-00/?p=20933) will required COM and the things that end up calling CreateProcess don't